backend/monitoring/dashboard.py

```
@@ -8,8 +8,60 @@
+        )
+from fastapi import HTTPException, status
+from fastapi.security import HTTPBearer, HTTPAuthorizationCredentials
+
+# Import proper admin authentication
+try:
+    from deps import get_current_admin_dependency
+except ImportError:
+    # If deps module is not available, create a secure fallback
+    security = HTTPBearer()
+    
+    async def get_current_admin_dependency(credentials: HTTPAuthorizationCredentials = Depends(security)):
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)